geniza/corpus/management/commands/import_data.py

```diff
@@ -311,46 +373,21 @@ def import_documents(self):
         joins = []
         docstats = defaultdict(int)
         for row in metadata:
-            if ";" in row.type:
-                logger.warning("skipping PGPID %s (demerge)" % row.pgpid)
-                docstats["skipped"] += 1
-                continue
-
-            doctype = self.get_doctype(row.type)
-            fragment = self.get_fragment(row)
-            doc = Document.objects.create(
-                id=row.pgpid,
-                doctype=doctype,
-                description=row.description,
+            joins, docstats = self.import_document(
+                row, joins, docstats, recto_verso_lookup
             )
-            doc.tags.add(*[tag.strip() for tag in row.tags.split("#") if tag.strip()])
-            # associate fragment via text block
-            TextBlock.objects.create(
-                document=doc,
-                fragment=fragment,
-                # convert recto/verso value to code
-                side=recto_verso_lookup.get(row.recto_verso, ""),
-                region=row.text_block,
-                subfragment=row.multifragment,
-            )
-            self.add_document_language(doc, row)
-            docstats["documents"] += 1
-            # create log entries as we go
-            self.log_edit_history(
-                doc, self.get_edit_history(row.input_by, row.date_entered, row.pgpid)
+
+        # update id sequence based on highest imported pgpid
+        self.update_document_id_sequence()
+
+        # demerged_metadata = reversed(sorted(demerged_metadata, key=lambda x: x.pgpid))
+        for row in demerged_metadata:
+            # overwrite document if it already exists
+            if row.pgpid:
+                Document.objects.filter(id=row.pgpid).delete()
```

Comment:
Documents in the demerge spreadsheet overwrite those in the metadata spreadsheet correct?

Reply:
I think we avoid ever having conflicts, because if we detect a document that needs demerging in the metadata spreadsheet, we don't add it to the database at all

Reply:
The assumption is that the demerge spreadsheet handles importing documents for all of those cases, and in fact we want to ensure that every demerge case detected from the main sheet is resolved via the demerge sheet